test/model/src/include/model/kv_database.h

```diff
@@ -101,6 +105,27 @@ class kv_database {
      */
     kv_checkpoint_ptr checkpoint(const char *name = nullptr);
 
+    /*
+     * kv_transaction::set_oldest_timestamp --
+     *     Set the database's oldest timestamp, if set.
+     */
+    inline void
+    set_oldest_timestamp(timestamp_t timestamp) noexcept
+    {
+        std::lock_guard lock_guard(_timestamps_lock);
+        _oldest_timestamp = std::max(_oldest_timestamp, timestamp);
```

Comment:
I'm not 100% sure that this matches WT behaviour - isn't it an error to try to make the oldest timestamp go backwards?

Reply:
It is an error, as described [here](https://github.com/wiredtiger/wiredtiger/pull/10427/files#diff-ba4c9e2fbaeaefdd722d5c7566e096e249c1ad3b59214e6b7443061ea9e9be2eR223).
Taking the max here means we should never have a lower value than the current `_oldest_timestamp`, it should be fine.

Reply:
When trying this empirically, this is the same behavior as I get with WT, and I added unit tests to make sure.

It is not currently an error for the oldest timestamp to be set to an earlier value, just like with the stable timestamp (at least until Etienne's PR gets merged in). I still didn't want to generate that behavior in the workload generator, which is why I added the check.